packages/compiler/core/semantic-walker.ts

```diff
@@ -238,6 +238,30 @@ function navigateType(
   }
 }
 
+// Find the immediate children of the given model.
+export function getChildModels(program: Program, parent: ModelType) {
```

Comment:
this looks good but I am a bit woried that if we have that like that it would lead to believe this is "fast" and abuse of this function would have bad performance for a very large spec.

Reply:
not sure how else you could share this functionality though

Reply:
I was wondering if we could do something like make a Map<Model, Model[]> in one pass on first call and use map on subsequent calls, but this is complicated by the fact that programs are mutable and mutations (new types, for example, would invalidate the map.)

Reply:
Maybe a verb that is more suggestive of the expense than `get` and a comment about the cost?

Reply:
Are we calling this on all models in our use? Did we make emit O(N^2) on the number of models?

Reply:
I think in the current usage it only gets called for models with discriminators but it could be abused on all models and then becomes(`n^2`)
For the naming maybe `findChildModels` instead.

Reply:
`findChildModels` is a better name.  I'll make that change.